pkg/network/master/master.go

```diff
@@ -146,6 +146,27 @@ func Start(networkConfig osconfigapi.MasterNetworkConfig, networkClient networkc
 		return err
 	}
 
+	var nodes []kapi.Node
+	var subnets []networkapi.HostSubnet
+	if nodeList, err := master.kClient.Core().Nodes().List(metav1.ListOptions{}); err == nil {
+		nodes = nodeList.Items
+	}
+	if subnetList, err := master.networkClient.Network().HostSubnets().List(metav1.ListOptions{}); err == nil {
+		subnets = subnetList.Items
+	}
+
+	for _, subnet := range subnets {
+		for i, node := range nodes {
+			if subnet.Name == node.Name {
+				break
+			}
+			if i == len(nodes)-1 {
```

Comment:
Instead of checking `if i == len(nodes)-1` in this loop, set a bool if you find a corresponding node and check the bool outside this loop and delete the subnet if required.